walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,75 +21,64 @@
-      return possibleTimes;
+      // If there are no events schedules, the whole day is available.
+      return Arrays.asList(WHOLE_DAY);
     }
-    return getPossibleTimes(events, attendees, optionalAttendees, request.getDuration());
-    
-  }
 
-  /**
-  * Returns an ArrayList of all possible meeting times that are free for the given list of attendees,
-  * and includes optional attendees if possible. 
-  *
-  * @param  events the Collection of Event objects
-  * @param  requestAttendees the Set of mandatory attendees
-  * @param  optAttendees the Set of optional attendees
-  * @param  duration the length of the requested meeting
-  * @return    the ArrayList of all possible meeting times that work with the attendees schedules
-  */
-  private ArrayList<TimeRange> getPossibleTimes(Collection<Event> events, Set<String> mandatoryAttendees,
-          Set<String> optAttendees, long duration) {
-    ArrayList<TimeRange> mandatoryBusyTimes = getBusyTimes(events, mandatoryAttendees);
-    ArrayList<TimeRange> optionalBusyTimes = getBusyTimes(events, optAttendees);
-    return timesWithAccountForOptAttendees(mandatoryBusyTimes, optionalBusyTimes, duration);
-  }   
+    Set<String> mandatoryAttendees =  new HashSet<String>(request.getAttendees());
+    Set<String> optionalAttendees =  new HashSet<String>(request.getOptionalAttendees());
+
+    return getPossibleTimes(getBusyTimes(events, mandatoryAttendees), 
+        getBusyTimes(events, optionalAttendees), request.getDuration());
+  }
 
   /**
-  * Returns an ArrayList of possible meeting times that prioritize the schedules of 
-  * mandatory meeting attendees and takes into account the schedule of optional attendees 
-  * if possible.
+  * Returns an ArrayList of all possible meeting times. The schedules of mandatory attendees
+  * are considered first with top priority. The schedule of optional attendees are considered
+  * when possible.
   *
   * @param  mandatory the ArrayList of timeranges where mandatory attendees are busy
   * @param  optional the ArrayList of timeranges where optional attendees are busy
   * @param  duration the length of the requested meeting
   * @return    the ArrayList of all open meeting times for mandatory attendees & if possible, optional attendees
   */
-  private ArrayList<TimeRange> timesWithAccountForOptAttendees(ArrayList<TimeRange> mandatoryTimes,
+  private ArrayList<TimeRange> getPossibleTimes(ArrayList<TimeRange> mandatoryTimes,
           ArrayList<TimeRange> optionalTimes, long duration) {
     if (mandatoryTimes.isEmpty() && !optionalTimes.isEmpty()) {
+      //If there are NO mandatory attendees but there ARE optional attendees
       return findFreeTimes(getOverlap(optionalTimes), duration);
     }
+    
     if (!mandatoryTimes.isEmpty() && optionalTimes.isEmpty()) {
+      //If there ARE mandatory attendees but NO optional attendees.
       return findFreeTimes(getOverlap(mandatoryTimes), duration);
     }
```

Comment:
I believe the following lines could be further simplified as the following:

```suggestion
    if (mandatoryTimes.isEmpty()) {
      return findFreeTimes(getOverlap(optionalTimes), duration);
    } else if (optionalTimes.isEmpty()) {
      return findFreeTimes(getOverlap(mandatoryTimes), duration);
    }
```

Reply:
Great suggestion, thanks!